flocker/node/test/test_script.py

```diff
@@ -58,29 +59,33 @@ def setup_config(test, control_address=u"10.0.0.1", control_port=1234,
     :param int control_port: The port number of the control service.
     :param unicode name: The ZFS pool name.  If ``None``, a random one will be
         generated that identifies the given test.
+    :param dict log_config: A logging configuration dictionary. If ``None``,
+        no logging stanza will be added to the configuration file.
     """
     if name is None:
         name = random_name(test)
     ca_set = get_credential_sets()[0]
     scratch_directory = FilePath(test.mktemp())
     scratch_directory.makedirs()
+    contents = {
+        u"control-service": {
+            u"hostname": control_address,
+            u"port": control_port,
+        },
+        u"dataset": {
+            u"backend": u"zfs",
+            u"name": name,
+            u"mount_root": scratch_directory.child(b"mount_root").path,
+            u"volume_config_path": scratch_directory.child(
+                b"volume_config.json"
+            ).path,
+        },
+        u"version": 1,
+    }
+    if log_config is not None:
+        contents[u'logging'] = log_config
     test.config = scratch_directory.child('dataset-config.yml')
-    test.config.setContent(
-        yaml.safe_dump({
-            u"control-service": {
-                u"hostname": control_address,
-                u"port": control_port,
-            },
-            u"dataset": {
-                u"backend": u"zfs",
-                u"name": name,
-                u"mount_root": scratch_directory.child(b"mount_root").path,
-                u"volume_config_path": scratch_directory.child(
-                    b"volume_config.json"
-                ).path,
-            },
-            u"version": 1,
-        }))
+    test.config.setContent(yaml.safe_dump(contents))
     ca_set.copy_to(scratch_directory, node=True)
     test.ca_set = ca_set
     test.non_existent_file = scratch_directory.child('missing-config.yml')
```

Comment:
:question: Probably shouldn't be done in this branch, but not a huge fan of setting attributes on the test object. When looking at an individual test, it makes it much harder to see where the fixtures are coming from (http://xunitpatterns.com/Obscure%20Test.html#Mystery Guest).